packages/custom-functions-metadata/test/cases/input-parameter-types-cellvalue/functions.js

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+/**
+ * Test Excel.CellValue parameter
+ * @param {Excel.CellValue} x
+ * @customfunction
+ */
+function testCellValueParam(x) {
+    // empty
+}
+
+/**
+ * Test Excel.BooleanCellValue parameter
+ * @param {Excel.BooleanCellValue} x
+ * @customfunction
+ */
+function testBooleanCellValueParam(x) {
+    // empty
+}
+
+/**
+ * Test Excel.DoubleCellValue parameter
+ * @param {Excel.DoubleCellValue} x
+ * @customfunction
+ */
+function testDoubleCellValueParam(x) {
+    // empty
+}
+
+/**
+ * Test Excel.EntityCellValue parameter
+ * @param {Excel.EntityCellValue} x
+ * @customfunction
+ */
+function testEntityCellValueParam(x) {
+    // empty
+}
+
+/**
+ * Test Excel.ErrorCellValue parameter
+ * @param {Excel.ErrorCellValue} x
+ * @customfunction
+ */
+function testErrorCellValueParam(x) {
+    // empty
+}
+
+/**
+ * Test Excel.FormattedNumberCellValue parameter
+ * @param {Excel.FormattedNumberCellValue} x
+ * @customfunction
+ */
+function testFormattedNumberCellValueParam(x) {
+    // empty
+}
+
+/**
+ * Test Excel.LinkedEntityCellValue parameter
+ * @param {Excel.LinkedEntityCellValue} x
+ * @customfunction
+ */
+function testLinkedEntityCellValueParam(x) {
+    // empty
+}
+
+/**
+ * Test Excel.LocalImageCellValue parameter
+ * @param {Excel.LocalImageCellValue} x
+ * @customfunction
+ */
+function testLocalImageCellValueParam(x) {
+    // empty
+}
+
+/**
+ * Test Excel.StringCellValue parameter
+ * @param {Excel.StringCellValue} x
+ * @customfunction
+ */
+function testStringCellValueParam(x) {
+    // empty
+}
+
+/**
+ * Test Excel.WebImageCellValue parameter
+ * @param {Excel.WebImageCellValue} x
+ * @customfunction
+ */
+function testWebImageCellValueParam(x) {
+    // empty
+}
+
+/**
+ * Test Excel.BooleanCellValue repeating parameter
+ * @param {Excel.BooleanCellValue[]} x
+ * 
+ * @customfunction
+ */
+function testBooleanCellValueRepeatingParam(x) {
+    // empty
+}
+
+/**
+ * Test Excel.BooleanCellValue matrix parameter
+ * @param {Excel.BooleanCellValue[][]} x
+ * @customfunction
+ */
+function testBooleanCellValueMatrixParam(x) {
+    // empty
+}
+
+/**
+ * Test Excel.BooleanCellValue repeating matrix parameter
+ * @param {Excel.BooleanCellValue[][][]} x
```

Comment:
Just curious when could CF have this kind of array? As data in the worksheet is 2d, we considering user just give input manually?

Reply:
I didn't know this before either, but apparently [][][] can be used for repeating range parameters. See https://learn.microsoft.com/en-us/office/dev/add-ins/excel/custom-functions-parameter-options?tabs=javascript .

Reply:
https://learn.microsoft.com/en-us/office/dev/add-ins/excel/custom-functions-parameter-options?tabs=javascript#declaring-repeating-parameters

Does custom-functions-metadata support repeating scalar or matrix parameters?  The help mentions hand-writing the functions.json metadata to have `repeating: true`. It would be good to support it so it can be determined from the dimensionality of the array type, e.g. `number[][][]`. This would not be for this PR but something else to add to the backlog.

Reply:
It is supported.  

https://github.com/OfficeDev/Office-Addin-Scripts/blob/fb1b94d0e3654680aa67088d098f4719e94484cd/packages/custom-functions-metadata/test/cases/repeating-param-3d/functions.ts#L5